src/main/java/taxi/controller/manufacturer/AddManufacturerController.java

```diff
@@ -21,11 +21,17 @@ public void doGet(HttpServletRequest req, HttpServletResponse resp)
     }
 
     @Override
-    public void doPost(HttpServletRequest req, HttpServletResponse resp) throws IOException {
+    public void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws IOException, ServletException {
         String name = req.getParameter("name");
         String country = req.getParameter("country");
+        if (name == null || name.length() == 0 || country == null || country.length() == 0) {
+            req.setAttribute("errorMsg", "Invalid name or country!");
+            req.getRequestDispatcher("/WEB-INF/views/manufacturers/add.jsp").forward(req, resp);
+        }
```

Comment:
Let's separate the name and country checks to not make the user guess which field was invalid